### PR TITLE
Add VESSL AI OpenAI API-compatible model provider and LLM model

### PR DESCRIPTION
**Pull request by larcane97**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. Close issue syntax: `Fixes #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

- add new vessl ai model provider 
- add open api url postfix in `OAIAPICompatLargeLanguageModel`
  - ref : effccc64bb791356baf0f5dc6ae2b1cc71143a8e

🔔  Not Yet added `yaml` configuration. It'll be added within a week.
- after adding, this comment will be deleted

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] test_validate_credentials
- [x] test_invoke_model
- [x] test_invoke_stream_model
- [x] test_get_num_tokens

### CODE REVIEW

File: api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py [hunk 1]
```diff
@@ -120,6 +120,8 @@ def validate_credentials(self, model: str, credentials: dict) -> None:
             endpoint_url = credentials["endpoint_url"]
             if not endpoint_url.endswith("/"):
                 endpoint_url += "/"
+            if not endpoint_url.endswith("v1/"):
+                endpoint_url += "v1/"
```
Comment (Yeuoly) [inline api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py]: Hi, It's not recommended to hard code path "v1" into `openai_api_compatible`, as it's not backwards compatible, all credentials already saved before will be appended with "v1", once it is originally it does not need "v1" as a prefix, their providers may not work anymore.

**Review by Yeuoly — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py (hunk 1 above)
Comment (larcane97) [reply api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py]: @Yeuoly
thank you for advice.
I reverted that part of code.

**Comment by Yeuoly** [comment]
It seems that you forgot to add the yaml file? This is necessary, otherwise frontend will not be able to render the icon and display the model name and other information as expected.

**Comment by larcane97** [comment]
@Yeuoly 
yes right as I mentioned above, the yaml file isn't applied yet.
I'll apply yaml file as soon as possible
I just wanted to ask you code level review !
once applying yaml file, I'll notify you again. thank you :)

**Comment by larcane97** [comment]
@Yeuoly
Hello Yeouly, i added yaml file and assets
then i think all i have to add is done, so can you review this pr?

**Review by laipz8200 — APPROVED** [review]

**Review by YIXIAO0 — APPROVED** [review]

**Review by crazywoola — APPROVED** [review]

**Pull request merged**